src/client/actions/MessageUpdate.js

```diff
@@ -32,6 +32,29 @@ class MessageUpdateAction extends Action {
       updated: null,
     };
   }
+
+  patchDataPacket(data, message) {
+    data.type = data.hasOwnProperty('type') ? data.type : Constants.MessageTypes.indexOf(message.type);
```

Comment:
duude

`indexOf` is for arrays and strings 😂, Constants.MessageTypes was defined as an `Object`

why not just use the good old index[] operator or, if paranoid, go with another ternary operator with hasOwnProperty?